README.md

```diff
@@ -17,30 +38,146 @@ For an always-on ML platform, the key requirement is the ability to train or ret
-   ```
-4. To install dependencies:
+3. To install dependencies:
     ```
     poetry install
     ```
-5. To run tests with coverage:
+4. To run unit tests:
     ```
     make test
     ```
-6. To format code style using black:
+5. To format code style using black:
     ```
     make format
     ```
+
+## Usage
+The following example shows a series of operations that can be performed using the package.
+### Generate synthetic data
+Let's start with generating some synthetic time series data using numalogic's synthetic module.
+```python
+from numalogic.synthetic import SyntheticTSGenerator
+
+ts_generator = SyntheticTSGenerator(
+    seq_len=8000,
+    num_series=3,
+    freq="T",
+    primary_period=720,
+    secondary_period=6000,
+    seasonal_ts_prob=0.8,
+    baseline_range=(200.0, 350.0),
+    slope_range=(-0.001, 0.01),
+    amplitude_range=(10, 75),
+    cosine_ratio_range=(0.5, 0.9),
+    noise_range=(5, 15),
+)
+ts_df = ts_generator.gen_tseries()  # shape: (8000, 3) with column names [s1, s2, s3]
+
+# Split into test and train
+train_df, test_df = ts_generator.train_test_split(ts_df, test_size=1000)
+```
+![Train and Test sets](images/train_test.png)
+
+### Generate synthetic anomalies
+Now we will add "contextual" anomalies in the test set, using numalogic's AnomalyGenerator.
+Contextual is one of the different types of anomalies that can be imputed. Others include "point", "collective"
+and "causal".
+```python
+from numalogic.synthetic import AnomalyGenerator
+
+injected_cols = ["s1", "s2"]  # columns to inject anomalies
+anomaly_generator = AnomalyGenerator(
+    train_df, anomaly_type="contextual", anomaly_ratio=0.3
+)
+outlier_test_df = anomaly_generator.inject_anomalies(
+    test_df, cols=injected_cols, impact=1.5
+)
+```
+![Outliers](images/outliers.png)
+
+### Model training 
+Let's use one of the available models to train on generated data.
+Here we are going to use a sparse convolutional Autoencoder.
+We can start by scaling the data.
+```python
+from sklearn.preprocessing import MinMaxScaler
+from numalogic.models.autoencoder.variants import Conv1dAE
+from numalogic.models.autoencoder.pipeline import SparseAEPipeline
+
+# Scale the train and test sets.
+scaler = MinMaxScaler()
+X_train = scaler.fit_transform(train_df.to_numpy())
+
+# Define the model and train
+seq_len = 36
+# SparseAEPipeline is a sparse autoencoder trainer that follows sklearn's API pattern
+model = SparseAEPipeline(
+    model=Conv1dAE(in_channels=3, enc_channels=8), seq_len=seq_len, num_epochs=30
+)
+model.fit(X_train)
+```
+
+### Model saving
+Now that the model is trained, let's save it. Numalogic has built in support 
+for Mlflow's tracking and logging system.
+```python
+from numalogic.registry import MLflowRegistrar
+
+# static and dynamic keys are used to look up a model
+static_keys = ["synthetic", "3ts"]
+dynamic_keys = ["minmaxscaler", "sparseconv1d"]
+
+registry = MLflowRegistrar(tracking_uri="http://0.0.0.0:8080", artifact_type="pytorch")
```

Comment:
we never documented how to start MLFlow server. may be we should add that too in `examples/` folder? Else the readme will not be complete.

Reply:
Yes, that's fair. We don't have an `examples` folder yet. But we can add one.

Reply:
Here is an elaborate link that we should point community to for this: https://www.mlflow.org/docs/latest/tracking.html#scenario-1-mlflow-on-localhost

Reply:
Referenced that link

Reply:
@s0nicboOm, I do not see any installation steps in that link.